translator/util/ecsutil/ecsutil.go

```diff
@@ -97,7 +103,18 @@ func (e *ecsUtil) parseRegion(em *ecsMetadataResponse) {
 	splitedContent := strings.Split(em.TaskARN, ":")
 	// When splitting the ARN with ":", the 4th segment is the region
 	if len(splitedContent) < 4 {
-		log.Printf("E! invalid ecs task arn: %s", em.TaskARN)
+		log.Printf("E! Invalid ecs task arn: %s", em.TaskARN)
 	}
 	e.Region = splitedContent[3]
 }
+
+// There is only one format for ClusterArn (https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_Cluster.html)
+// arn:aws:ecs:region:aws_account_id:cluster/cluster-name
+func (e *ecsUtil) parseClusterName(em *ecsMetadataResponse) {
+	splitedContent := strings.Split(em.Cluster, "/")
+	// When splitting the ClusterName with /, the last is always the cluster name
+	if len(splitedContent) == 0 {
+		log.Printf("E! Invalid cluster arn: %s", em.Cluster)
+	}
```

Comment:
If the ARN is invalid, should we still try to assign the cluster ARN to the last element?

Reply:
Never mind. I see it in the old code. I would feel better if it was ported over more accurately to the existing code in `get_ecs_cluster_name`

Reply:
I wrote this code based on behavior of [parsing Region from taskARN](https://github.com/aws/amazon-cloudwatch-agent/pull/453/files#diff-8b21e9e828737c79625db57c62d2f87fabcd2e60434cf9b1c4573e4c15da5324R102-R109). However, I have not had more information regarding this to change behavior. However, for one short notice is that even though the `clusterName` is invalid based on what I have skimmed, it would not affect the whole flow so I am still considering whether we need to fatal if clusterName is invalid.